src/main/java/com/alysaa/geyserupdater/bungee/util/GeyserBungeeDownloader.java

```diff
@@ -26,7 +26,13 @@ public static boolean updateGeyser() {
 
         // Download the file
         plugin.getProxy().getScheduler().runAsync(plugin, () -> {
-            String fileUrl = "https://ci.opencollab.dev/job/GeyserMC/job/Geyser/job/master/lastSuccessfulBuild/artifact/bootstrap/bungeecord/target/Geyser-BungeeCord.jar";
+            String fileUrl = null;
+            try {
+                fileUrl = "https://ci.opencollab.dev/job/GeyserMC/job/Geyser/job/" + GeyserProperties.getGeyserGitPropertiesValueForPropertyKey("git.branch") + "/lastSuccessfulBuild/artifact/bootstrap/bungeecord/target/Geyser-BungeeCord.jar";
```

Comment:
Can we store the branch on server startup to avoid querying the resource so much? also applies to the other platforms

Reply:
I'll work on this — added to the to-do list on the original post.

Reply:
I guess this'll be addressed in the next PR, as discussed on Discord.